terraform/modules/critical/s3_replica_glacier.tf

```diff
@@ -16,18 +16,44 @@ resource "aws_s3_bucket" "replica_glacier" {
   }
 }
 
-resource "aws_s3_bucket_policy" "replica_glacier_read" {
-  count = length(var.replica_glacier_read_principals) == 0 ? 0 : 1
-
+resource "aws_s3_bucket_policy" "replica_glacier_access" {
   bucket = aws_s3_bucket.replica_glacier.id
-  policy = data.aws_iam_policy_document.replica_glacier_readonly.json
+  policy = data.aws_iam_policy_document.replica_glacier_access.json
 }
 
-data "aws_iam_policy_document" "replica_glacier_readonly" {
+data "aws_iam_policy_document" "replica_glacier_access" {
+  # By default, we don't allow anybody to write or delete objects in the bucket.
+  #
+  # We only allow write permissions to certain, pre-approved IAM roles.  This is
+  # enforced at the bucket level, so even creating a role with "S3 full access" will
+  # not allow you to modify bucket objects.
```

Comment:
As reading from Glacier, especially in large data sizes, can be very very expensive, should we be restricting that too?

Reply:
The potential cost of pulling the entire cold copy out of storage is actually quite manageable.

This bucket is cycling all its objects to Glacier Deep Archive, which doesn't have expedited retrieval. The cost of retrieving the entire archive would be annoying but perfectly manageable:

$0.021 / GB * ~60 TB = ~$1300

https://aws.amazon.com/s3/pricing/

I don't think we'd want to do it very often, but it's not on the same tier of "never ever ever ever ever do this" as deleting or modifying the archive in-place.